frontend/src/lib/docs.ts

```diff
@@ -6,31 +6,41 @@ import Swagger from '@apidevtools/swagger-parser';
 import { OpenAPIV2 } from 'openapi-types';
 import { request } from './k8s/apiProxy';
 
-let docsPromise: ReturnType<typeof getDocs>;
+let docsPromise: ReturnType<typeof getDocs> | null = null;
 
 async function getDocs() {
   const docs = await request('/openapi/v2');
   return Swagger.dereference(docs);
 }
 
+export function resetDocsPromise() {
+  docsPromise = null;
+}
+
 export default async function getDocDefinitions(apiVersion: string, kind: string) {
   if (!docsPromise) {
     docsPromise = getDocs(); // Don't wait here. Just kick off the request
   }
 
-  const { definitions = {} } = (await docsPromise) as OpenAPIV2.Document;
-
-  let [group, version] = apiVersion.split('/');
-  if (!version) {
-    version = group;
-    group = '';
-  }
-
-  return Object.values(definitions)
-    .filter(x => !!x['x-kubernetes-group-version-kind'])
-    .find(x => x['x-kubernetes-group-version-kind'].some(comparer));
-
-  function comparer(info: OpenAPIV2.SchemaObject) {
-    return info.group === group && info.version === version && info.kind === kind;
+  try {
+    const { definitions = {} } = (await docsPromise) as OpenAPIV2.Document;
+
+    let [group, version] = apiVersion.split('/');
+    if (!version) {
+      version = group;
+      group = '';
+    }
+
+    return Object.values(definitions)
+      .filter(x => !!x['x-kubernetes-group-version-kind'])
+      .find(x => x['x-kubernetes-group-version-kind'].some(comparer));
+
+    function comparer(info: OpenAPIV2.SchemaObject) {
+      return info.group === group && info.version === version && info.kind === kind;
+    }
+  } catch (error) {
+    // Reset docsPromise on error so subsequent requests can try again
+    resetDocsPromise();
+    throw error;
```

Comment:
these changes were made in light of PR #2908 and are present over there as well. Specifically, I added this for the [ErrorDocumentation](https://github.com/headlamp-k8s/headlamp/blob/98f0a0897c40db76791498548c88eaaaac30f1b6/frontend/src/components/common/Resource/DocsViewer.stories.tsx#L108-#L124) storybook test scenario present in that PR.

Reply:
Outdated convo... This was updated already.